src/main/java/org/spongepowered/common/event/tracking/context/MultiBlockCaptureSupplier.java

```diff
@@ -561,6 +561,9 @@ public boolean processTransactions(List<Transaction<BlockSnapshot>> transactions
                 try (final SpongeProxyBlockAccess.Proxy transactionProxy = transaction.getProxy(mixinWorldServer)) {
                     transaction.process(eventTransaction, phaseState, phaseContext, currentDepth);
                 }
+                if (this.processingBlocks != null) {
+                    this.processingBlocks.remove(mixinWorldServer);
```

Comment:
No.... It needs to retain the proxy to close the proxy itself when it's completed.

Reply:
Yeah, no... You really don't want to remove in the for loop. There's a reason why it's being closed and pruned in the finally block below. The proxy created for the `BlockTransaction` is auto-closed by the try with resources above.